tests/native/test_models.py

```diff
@@ -262,6 +263,39 @@ def strlen_model(state):
         # Assert that every expected stdout has a matching output
         self.assertEqual(expected, stdouts)
 
+    def test_symbolic_fork_unique_solution(self):
+        # This binary is compiled using gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
+        # with flags: -g -static -fno-builtin
+        BIN_PATH = os.path.join(
+            os.path.dirname(__file__), "binaries/str_model_tests", "sym_strlen_test"
+        )
+        tmp_dir = tempfile.TemporaryDirectory(prefix="mcore_test_sym_")
+        m = Manticore(BIN_PATH, stdin_size=10, workspace_url=str(tmp_dir.name))
+
+        addr_of_strlen = 0x04404D0
+
+        @m.hook(addr_of_strlen)
+        def strlen_model(state):
+            stdin_1 = ArrayProxy(array=state.constraints.get_variable("STDIN")).select(1)
+            state.constrain(stdin_1 == 0)
+            state.invoke_model(strlen_exact)
+
+        m.run()
+        m.finalize()
+
+        # Expected stdout outputs
+        expected = {"Length of string is: 0", "Length of string is: 1"}
```

Comment:
Would we expect to see different results for this test before the changes to our forking strategy? If we're forking on the length of the string, and constrain the _second_ byte to be zero, then it makes sense that valid strings would be of length zero or one. That's two possibilities, which doesn't seem like it would test the new forking policy. 

Would it make more sense to inspect the fork events using a plugin, or inspect the state ID numbers to confirm that we aren't creating new states when we concretize a symbol with a single possible value?

Reply:
Yeah that's spot on. I hadn't noticed that the `strlen_exact` callback doesn't concretise the individual bytes but uses the solver API to check whether a byte is `null`, avoiding any forks. Definitely an oversight on my part.

I will replace that test with one that creates a Manticore instance and then raises a `ConcretizeException` on a constrained symbolic variable, then just checks that no additional states were created using `will_fork_state_callback`.

Reply:
I have replaced the test with another one here: https://github.com/trailofbits/manticore/blob/003ce0ad620046e9602e95c1eb2c763ba66cedf9/tests/other/test_fork.py 

The new test executes the simple `hello_world` program and artificially concretises a symbolic variable (constrained to one unique value) from a hook, and finally asserts that no additional states were created when forking. Let me know what you think!